src/docker-in-docker/install.sh

```diff
@@ -469,32 +469,24 @@ if [ "${INSTALL_DOCKER_BUILDX}" = "true" ]; then
     find "${docker_home}" -type d -print0 | xargs -n 1 -0 chmod g+s
 fi
 
-if [ -z "$DISABLE_IP6_TABLES" ]; then
-    DOCKER_DEFAULT_IP6_TABLES=""
-else
-    version_to_check=""
-    semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
-    if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
-        version_to_check=$(echo $DOCKER_VERSION | cut -d. -f1)
-    elif echo "$DOCKER_VERSION" | grep -Eq "^-?[0-9]+$"; then
-        version_to_check=$DOCKER_VERSION
-    fi
+DOCKER_DEFAULT_IP6_TABLES=""
+version_to_check=""
+semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
+if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
+    version_to_check=$(echo $DOCKER_VERSION | cut -d. -f1)
+elif echo "$DOCKER_VERSION" | grep -Eq "^-?[0-9]+$"; then
+    version_to_check=$DOCKER_VERSION
+fi
 
-    if [ -n "$version_to_check" ] && [ "$version_to_check" -ge 27 ] || [ "$DOCKER_VERSION" = "latest" ]; then
-            if [ "$DISABLE_IP6_TABLES" == false ]; then
-                DOCKER_DEFAULT_IP6_TABLES=""
-            else
-                echo "Info: Ip6tables feature is enabled by default in docker engine v27 & later. Disabling.."
-                DOCKER_DEFAULT_IP6_TABLES="--ip6tables=false"
-            fi
-    else 
+if [[ -n "$version_to_check" && "$version_to_check" -ge 27 ]] || [ "$DOCKER_VERSION" = "latest" ]; then
         if [ "$DISABLE_IP6_TABLES" == true ]; then
-            echo "Info: Ip6tables feature is already disabled in docker engine v26 & lower."
-            DOCKER_DEFAULT_IP6_TABLES=""
-        else
-            echo "Error: Ip6tables in docker engine v26 & lower is an experimental feature. Enabling is not supported."
-            exit 1
+            echo "Info: Ip6tables feature is enabled by default in docker engine v27 & later. Disabling.."
+            DOCKER_DEFAULT_IP6_TABLES="--ip6tables=false"
         fi
+else 
+    if [ "$DISABLE_IP6_TABLES" == false ]; then
```

Comment:
Shouldn't this be `true` ?

Reply:
actually, v26 and lower already has this disabled by default so testing true doesn't make sense, added a condition check if the user actually wanted it to be false i.e. trying to enable the setting of `--ip6tables=true`, only this condition makes sense for v26 and lower

Reply:
Is passing `--ip6tables` supported for docker v26 and lower?

Reply:
<img width="458" alt="image" src="https://github.com/user-attachments/assets/35ed9df7-8938-4ff3-b0e4-ce7a81d72ab2">

yeah it is supported as is written in the docs

Reply:
Got it, thanks for looking into it!  

So given the complimentary situation, let's only bother for v27 and up, and avoid conflicts and changing default behavior. 

Do nothing if `DOCKER_DEFAULT_IP6_TABLES=false`, and only pass `--ip6tables=true` if `DOCKER_DEFAULT_IP6_TABLES=true`

@gauravsaini04 Does that make sense?

Reply:
changed code